internal/backend/db/dbgorm/projects.go

```diff
@@ -59,11 +59,14 @@ func (db *gormdb) deleteProjectAndDependents(ctx context.Context, projectID stri
 
 	gormDB := db.db.WithContext(ctx)
 
-	// delete triggers
 	if err = gormDB.Delete(&scheme.Trigger{}, "project_id = ?", projectID).Error; err != nil {
 		return err
 	}
 
+	if err = gormDB.Delete(&scheme.Connection{}, "project_id = ?", projectID).Error; err != nil {
+		return err
+	}
+
```

Comment:
how do we know no one is referencing this connection ?
delete order ?

Reply:
Connection could be referenced by Triggers or Signals. Triggers are deleted before Connections. And now since it's a hard delete, if there is anyone referencing this connection, delete will fail on foreign key

Reply:
connection are also referenced by events
so this is something we should think about
not sure we can delete them now

Reply:
> Connection could be referenced by Triggers or Signals. Triggers are deleted before Connections. And now since it's a hard delete, if there is anyone referencing this connection, delete will fail on foreign key

Please add this information as a comment in the code. This will help future readers to understand the code and the flow much more quickly.